src/router/User.js

```diff
@@ -0,0 +1,66 @@
+import express from 'express';
+import joiValidator from 'express-joi-validation';
+import { randomUUID } from 'crypto';
+import users from '../data/users.json';
+import userSchema from '../validation/user';
+import { isLoginUniq } from '../util/user';
+
+const validator = joiValidator.createValidator({});
+
+const router = express.Router();
+
+router.param('id', (req, _res, next, id) => {
+  req.user = users.find(user => user.id === id);
+  if (req.user) {
+    next();
+  } else {
+    next(`User with id '${id}' not found`);
+  }
+});
```

Comment:
Interesting approach, but it's better to avoid any request modifications. After adding a logging you will expose sensitive information to log files. Rework it in the next homework